spec_42.txt

```diff
@@ -0,0 +1,179 @@
+
+You should have received a copy of the GNU General Public License along with this program; if not, see <http://www.gnu.org/licenses>.
+
+++ Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [http://tools.ietf.org/html/rfc2119 RFC 2119].
+
+++ Goals
+
+C4 is meant to provide a reusable optimal collaboration model for open source software projects. It has these specific goals:
+
+* To maximize the scale and diversity of the community around a project, by reducing the friction for new Contributors and creating a scaled participation model with strong positive feedbacks;
+
+* To relieve dependencies on key individuals by separating different skill sets so that there is a larger pool of competence in any required domain;
+
+* To allow the project to develop faster and more accurately, by increasing the diversity of the decision making process;
+
+* To support the natural life cycle of project versions from experimental through to stable, by allowing safe experimentation, rapid failure, and isolation of stable code;
+
+* To reduce the internal complexity of project repositories, thus making it easier for Contributors to participate and reducing the scope for error;
+
+* To enforce collective ownership of the project, which increases economic incentive to Contributors and reduces the risk of hijack by hostile entities.
+
+++ Design
+
++++ Preliminaries
+
+* The project SHALL use the git distributed revision control system.
+
+* The project SHALL be hosted on github.com or equivalent, herein called the "Platform".
+
+* The project SHALL use the Platform issue tracker.
+
+* The project SHOULD have clearly documented guidelines for code style.
+
+* A "Contributor" is a person who wishes to provide a patch, being a set of commits that solve some clearly identified problem.
+
+* A "Maintainer" is a person who merges patches to the project. Maintainers are not developers; their job is to enforce process.
+
+* Contributors SHALL NOT have commit access to the repository unless they are also Maintainers.
+
+* Maintainers SHALL have commit access to the repository.
+
+* Everyone, without distinction or discrimination, SHALL have an equal right to become a Contributor under the terms of this contract.
+
++++ Licensing and Ownership
+
+* The project SHALL use a share-alike license, such as the GPLv3 or a variant thereof (LGPL, AGPL), or the MPLv2.
+
+* All contributions to the project source code ("patches") SHALL use the same license as the project.
+
+* All patches are owned by their authors. There SHALL NOT be any copyright assignment process.
+
+* The copyrights in the project SHALL be owned collectively by all its Contributors.
```

Comment:
Someone from the Linux foundation told the above 2 conflicts each other.

So if I made a patch, do I own the copyrights? Can I demand that the patch will be removed? Anyway, as said I don't understand this much, we might want to check it out.